core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -420,6 +435,28 @@ private static String toString(Collection<String> items) {
     }
 
     /**
+     * The lock number we were able to take. This is <strong>usually</strong> the number of Elasticsearch nodes sharing data directories
+     * with this node.
+     */
+    public int getNodeLockId() {
+        return nodeLockId;
+    }
+
+    /**
+     * Get the maximum number of local storage nodes allowed.
+     * @param settings the settings object from which to read the settings
+     * @param productionMode whether or not we are running in production mode
+     */
+    public static int getMaxLocalStorageNodes(Settings settings, boolean productionMode) {
+        int maxLocalStorageNodes = MAX_LOCAL_STORAGE_NODES_SETTING.get(settings);
```

Comment:
The confusing part is that I had it as a class variable at some point when I was working on it!